libs/remix-ui/file-explorer/src/lib/file-explorer.tsx

```diff
@@ -163,10 +166,7 @@ export const FileExplorer = (props: FileExplorerProps) => {
   useEffect(() => {
     if (contextMenuItems) {
       setState(prevState => {
-        // filter duplicate items
-        const items = contextMenuItems.filter(({ name }) => prevState.actions.findIndex(action => action.name === name) === -1)
-
-        return { ...prevState, actions: [...prevState.actions, ...items] }
+        return { ...prevState, actions: [...initialActions, ...contextMenuItems] }
```

Comment:
This might not work because in the copy & paste PR (https://github.com/ethereum/remix-project/pull/1209/files#diff-12461633b1b7400c1f8d4b9e2480ab50e052db7f4c71591121d2b47f8077c677R244, https://github.com/ethereum/remix-project/pull/1209/files#diff-12461633b1b7400c1f8d4b9e2480ab50e052db7f4c71591121d2b47f8077c677R253), context menu items can be added and removed during runtime for file-explorer. It will be better to have `{ ...prevState, actions: [...prevState.actions, ...contextMenuItems] }`. We still need to check for duplicates because the check for duplicate in file-panel is not sufficient. File-panel does not have access to `initialActions` and cannot prevent adding menus already present in `initailActions`.

Reply:
Ok sure, but there needs to be a way for items to be removed when contextMenuItems prop changes, in this case shrinks when a plugin is deactivated. however way that is done is fine. but likes this all the menuitems will remain after deactivation.

Reply:
Yeah, you're right. Currently, the `useEffect` for `contextMenuItems` makes use of `addMenuItems` https://github.com/ethereum/remix-project/pull/1209/files#diff-12461633b1b7400c1f8d4b9e2480ab50e052db7f4c71591121d2b47f8077c677R187. We can do the same for when items are removed since the file-panel manages the removal of menu items. We would just need to call `removeMenuItems` with an array of the ids.

Reply:
@ioedeveloper this is still an unresolved issue. So what do we do? I don't want to mess in your react code but I can make a suggestion if needed.